osu.Game/Rulesets/ILegacyRuleset.cs

```diff
@@ -13,5 +13,12 @@ public interface ILegacyRuleset
         /// Identifies the server-side ID of a legacy ruleset.
         /// </summary>
         int LegacyID { get; }
+
+        /// <summary>
+        /// For rulesets which support legacy (osu-stable) replay conversion, this method will create an empty replay frame
+        /// for conversion use.
+        /// </summary>
+        /// <returns>An empty frame for the current ruleset, or null if unsupported.</returns>
+        IConvertibleReplayFrame CreateConvertibleReplayFrame();
```

Comment:
I would've rather kept things separate by just marking this method as nullable in `Ruleset` rather than introducing API breakage. But sure I suppose?

Reply:
Also if this will still persist, it may be a good opportunity to move `ConvertFromLegacyMods` and `ConvertToLegacyMods` inside this interface as well.

Reply:
I guess I'll just roll back this change if need to merge it in the next release.